src/streamlit_app.py

```diff
@@ -87,14 +99,14 @@ def architecture_dialog() -> None:
                 "Prompts, responses and feedback in this app are anonymously recorded and saved to LangSmith for product evaluation and improvement purposes only."
             )
 
+        st.markdown(f"Chat ID: **{st.session_state.thread_id}**")
```

Comment:
Add something like `help="Set ?thread_id=… to retrieve this conversation"`  so it’s more obvious?

Reply:
Great idea, done!